### PR TITLE
Release 0.2.7

### PR DESCRIPTION
**Pull request by felipevaladares**
- Merge background and foreground listeners into a single feature
- Changed duplicates control to use anchors instead of local database for all queries that aren't statistical
- Refactored and simplified HealthQuery structures
- Updating historical and latest data sync
- Using new DNS
- General improvements

### CODE REVIEW

File: docs/CapacitorReferences.md [hunk 1]
```diff
@@ -14,18 +14,12 @@ This document contains references to all available methods and types generated b
...
-- [`startBackgroundListenersForType(...)`](#startbackgroundlistenersfortype)
-- [`disableAllBackgroundListeners()`](#disableallbackgroundlisteners)
-- [`disableBackgroundListenersForType(...)`](#disablebackgroundlistenersfortype)
-- [`enableAllForegroundListeners()`](#enableallforegroundlisteners)
-- [`enableForegroundListenerForType(...)`](#enableforegroundlistenerfortype)
-- [`stopAllForegroundListeners()`](#stopallforegroundlisteners)
-- [`stopForegroundListenerForType(...)`](#stopforegroundlistenerfortype)
+- [`startAllListeners()`](#startalllisteners)
+- [`startListenerForType(...)`](#startlistenerfortype)
+- [`StopAllListeners()`](#stopalllisteners)
```
Comment (AnnyFigueira) [inline docs/CapacitorReferences.md:19]:
```suggestion
- [`stopAllListeners()`](#stopalllisteners)
```

File: docs/CapacitorReferences.md [hunk 2]
```diff
@@ -180,81 +174,25 @@ Enable background listeners to sync just a specific query type.
 
 ---
 
-### disableAllBackgroundListeners()
+### StopAllListeners()
 
 ```typescript
-disableAllBackgroundListeners() => Promise<any>
+StopAllListeners() => Promise<any>
```
Comment (AnnyFigueira) [inline docs/CapacitorReferences.md:180]:
```suggestion
stopAllListeners() => Promise<any>
```

File: docs/CapacitorReferences.md [hunk 3]
```diff
@@ -180,81 +174,25 @@ Enable background listeners to sync just a specific query type.
 
 ---
 
-### disableAllBackgroundListeners()
+### StopAllListeners()
```
Comment (AnnyFigueira) [inline docs/CapacitorReferences.md:177]:
```suggestion
### stopAllListeners()
```

File: src/definitions.ts [hunk 4]
```diff
@@ -58,44 +58,24 @@ export interface PointSDKPlugin {
...
    */
-  startBackgroundListenersForType(options: { type: QueryType }): Promise<any>;
+  startListenerForType(options: { type: QueryType }): Promise<any>;
 
   /**
-   * Disables background listeners for all types defined on SDK setup.
+   * Stop listeners for all types defined on SDK setup.
    */
-  disableAllBackgroundListeners(): Promise<any>;
+  StopAllListeners(): Promise<any>;
```
Comment (AnnyFigueira) [inline src/definitions.ts:73]:
```suggestion
  stopAllListeners(): Promise<any>;
```

**Review by AnnyFigueira — CHANGES_REQUESTED** [review]

**Review by GGGava — APPROVED** [review]
Apart from Anny's comments, lgtm

**Review by AnnyFigueira — APPROVED** [review]

**Pull request merged**